php/ext/google/protobuf/message.c

```diff
@@ -754,16 +755,25 @@ PHP_METHOD(Message, serializeToJsonString) {
   size_t size;
   int options = 0;
   char buf[1024];
-  zend_bool preserve_proto_fieldnames = false;
+  zval *flags = NULL;
   upb_Status status;
 
-  if (zend_parse_parameters(ZEND_NUM_ARGS(), "|b",
-                            &preserve_proto_fieldnames) == FAILURE) {
+  if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z",
+                            &flags) == FAILURE) {
     return;
   }
 
-  if (preserve_proto_fieldnames) {
+  if (flags == NULL || Z_TYPE_P(flags) == IS_FALSE) {
```

Comment:
FYI I've left this boolean-handling code in the C code, as `Z_LVAL_P(bool)` causes a segfault (and the param needs to be a zval to accept both int&bool inputs)

Reply:
I don't know C++ well, but I added a bunch more test cases to ensure that this functions as expected and aligns with the previous behavior (I added tests for boolean `true`, `false`, and calling without parameters). So I feel confident that this behavior is being preserved, and it should be okay.

Thanks for fixing the segfault :)